modules/subscriptions/subscriptions/subscriptions.test.tsx

```diff
@@ -0,0 +1,197 @@
+    });
+
+    it("Checks subscribe button on second price card redirects", async () => {
+      const mockGetSubscriptionLink = jest
+        .fn<() => Promise<{ url: string }>>()
+        .mockResolvedValue({ url: "https://example.com/subscribe" });
+
+      const serviceGetSpy = jest.spyOn(Service, "get").mockReturnValue({
+        getSubscriptionLink: mockGetSubscriptionLink,
+      } as any);
+
+      render(<Subscriptions />);
+      const subscribeButton = await screen.findByText(
+        "subscriptions.price-2.btn-text",
+      );
+      await userEvent.click(subscribeButton);
+
+      expect(mockGetSubscriptionLink).toHaveBeenCalled();
+
+      await waitFor(() => {
+        expect(window.location.href).toBe("https://example.com/subscribe");
+      });
+
+      serviceGetSpy.mockRestore();
+    });
+  });
+
+  describe("Subscriptions Settings Page", () => {
+    let Subscriptions: CoreComponent;
+
+    beforeAll(() => {
+      Subscriptions = getComponent("settings", "subscriptions");
+    });
+
+    beforeEach(() => {});
+    afterEach(() => {});
+
+    it("Subscription settings is Mounted", async () => {
+      await act(() => render(<Subscriptions />));
+
+      expect(
+        screen.getByTestId("subscriptions-settings.main"),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByText("subscription-settings.cancel-sub-btn"),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByText("subscription-settings.payment-history"),
+      ).toBeInTheDocument();
+      screen.debug();
+    });
+
+    it("Checks payment history is loaded", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("payment-history.main")).toBeInTheDocument();
+        expect(screen.getByTestId("payment-history.main")).toBeInTheDocument();
+      });
+    });
+
+    it("Checks table is populated", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+    });
+
+    it("Checks view receipt link redirects", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const viewReceiptLink = await screen.findAllByText(
+        "invoice-history.view-receipt",
+      );
+      expect(viewReceiptLink[0]).toBeInTheDocument();
+      expect(viewReceiptLink[0]).toHaveAttribute(
+        "href",
+        expect.stringMatching(/^https:\/\/.+/),
+      );
+      expect(viewReceiptLink[0]).toHaveAttribute("target", "__blank");
+    });
+
+    it.todo("Checks cancel subscription button redirects");
+    it("Checks cancel subscription button redirects", async () => {
+      render(<Subscriptions />);
+
+      const cancelSubBtn = await screen.findByText(
+        "subscription-settings.cancel-sub-btn",
+      );
+      act(() => cancelSubBtn.click());
+
+      expect(cancelSubBtn).toBeInTheDocument();
+    });
+  });
```

Comment:
Test that the date of the subscription is displayed.
Add another test is the same when the status is cancel.
Check what happens when is expired, if we display the page properly (What message error comes)

Reply:
@Mrtn-Tianlu  We don't create a subscription, we use the mocked service to return a subscription on the api :)
Not sure this answer your question.
The test won't connect with slack